gramps/gen/db/select_utils.py

```diff
@@ -0,0 +1,315 @@
+#
+# Gramps - a GTK+/GNOME based genealogy program
+#
+# Copyright (C) 2024 Doug Blank <doug.blank@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+"""
+Support functions for Generic DB select functionality
+"""
+
+import re
+import jsonpath_ng
+
+PARSE_CACHE = {}
+OPERATORS = ("=", "!=", "<", "<=", ">", ">=", "in", "not in", "like")
+
+
+def select(db, table, selections, where, sort_by, page, page_size):
+    """This is a DB-API implementation of the DbGeneric.select()
+    method.
+
+    :param table: Name of table
+    :type table: str
+    :param selections: List of json-paths
+    :type selections: List[str] or tuple(str)
+    :param where: A single where-expression (see below)
+    :type where: tuple or list
+    :param sort_by: A list of expressions to sort on
+    :type where: tuple or list
+    :param page: The page number to return (zero-based)
+    :type page: int
+    :param page_size: The size of a page in rows; None means ignore
+    :type page: int or None
+    :returns: Returns selected items from select rows, from iterator
+    :rtype: dict
+
+    Examples:
+
+    Get the gender and surname of all males, sort by gramps_id:
+    ```
+    db.select(
+        "person",
+        ["$.gender", "$.primary_name.surname_list[0].surname"],
+        where=["$.gender", "=", Person.MALE],
+        sort_by=["$.gramps_id"],
+    )
+    ```
+    Notes:
+
+    Although the SQL engine may support more variations than
+    Python (or other implementations) you should not use them to
+    ensure that your code will run with all backends.
+
+    The where expressions only support the following operators:
+        "=", "!=", "<", "<=", ">", ">=", "in", "not in", "like"
+
+    The `like` operator supports "%" (zero or more characters)
+    and "_" (one character) regular expression matches.
+    """
+    selections = selections if selections else ["$"]
+    if page_size is None:
+        limit = float("+inf")
```

Comment:
```suggestion
        limit = sys.maxsize
```
Could you use sys.maxsize instead of converting a string to a float?